### PR TITLE
Add Unit Tests for OkHttpJsonApiClient:getCampaigns

### LINKED ISSUE #2819: Add unit test for fetching campaigns
**Summary:** 

Now that we have #2793 in place we can easily test campaigns. At least we should have unit tests to verify the API layer (using mock web server). Specifically we need to test `OkHttpJsonApiClient:getCampaigns` for 2 scenarios. 

- when the API returns at least 1 campaign(active or otherwise)
- when the API returns no campaigns.

### PR DESCRIPTION
**Pull request by SaumiaSinghal**
**Description (required)**

Fixes #2819

What changes did you make and why?
I used `MockWebServer` to test `OkHttpJsonApiClient:getCampaigns`.

**Tests performed (required)**

**Screenshots (for UI changes only)**

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

File: app/src/test/kotlin/fr/free/nrw/commons/mwapi/OkHttpJsonApiClientTest.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        campaign.doOnSuccess { assertNull(it) }
+    }
+
+    @Test
+    fun whenApiHasCampaigns_getCampaignShouldReturnCampaigns() {
+        val baseUrl: HttpUrl = mockWebServer.url("/api/blah");
+
+        val json = TestFileUtil.readRawFile("campaigns_beta_active.json")
+        mockWebServer.enqueue(MockResponse().setBody(json.toString()));
+
```
Comment (SaumiaSinghal) [inline app/src/test/kotlin/fr/free/nrw/commons/mwapi/OkHttpJsonApiClientTest.kt:73]: To test for the case when the API returns an active campaign, I enqueue json file on `mockWebServer`. However, I'm getting an error on running this test, it says cannot find `org.wikipedia.test.TestFileUtil`.

Comment (maskaravivek) [reply app/src/test/kotlin/fr/free/nrw/commons/mwapi/OkHttpJsonApiClientTest.kt:73]: @ashishkumar468 Can you take a look at this issue? I remember that you helped me fix a similar issue some time back.

**Comment by neslihanturan** [comment]
Hey @SaumiaSinghal , are you still around and interested to work on this PR?

**Comment by neslihanturan** [comment]
Closing this one since it seems sleeping for a while, please feel free to reopen or inspire from this solution.

**Comment by shankarpriyank** [comment]
Hey, @nicolas-raoul I have fixed the error in this pr. Can I make a new pr fixing the issue?

**Comment by nicolas-raoul** [comment]
@shankarpriyank Yes thanks! Please comment on the issue.

**Pull request closed without merging**